src/dbnode/topology/map.go

```diff
@@ -46,6 +48,8 @@ func NewStaticMap(opts StaticOptions) Map {
 		shardSet:                 opts.ShardSet(),
 		hostShardSets:            hostShardSets,
 		hostShardSetsByID:        make(map[string]HostShardSet),
+		leavingHostMap:           make(map[string]map[uint32]string),
```

Comment:
before we go adding the memory for this - lets validate we need it. add to the list of things to check in real world benchmarks when you run against a cluster.